### ISSUE TITLE
🚨 Potential Security Vulnerability

### ISSUE THREAD

**Issue by ranjit-git** [projectsend/projectsend#962]
Hello, - 
potential high severity security vulnerability in your repository has been disclosed to huntr.
Visit report url and validate them 
https://huntr.dev/bounties/1625188746038-projectsend/projectsend/
https://huntr.dev/bounties/1625188855252-projectsend/projectsend/
https://huntr.dev/bounties/1625195879034-projectsend/projectsend/
https://huntr.dev/bounties/1625196531585-projectsend/projectsend/
https://huntr.dev/bounties/1625197455990-projectsend/projectsend/
https://huntr.dev/bounties/1625198773635-projectsend/projectsend/
https://huntr.dev/bounties/1625227550260-projectsend/projectsend/
https://huntr.dev/bounties/1625228023793-projectsend/projectsend/
https://huntr.dev/bounties/1625228365696-projectsend/projectsend/
https://huntr.dev/bounties/1625228789050-projectsend/projectsend/
Report details is only visible to repo-maintainer and reporter 
You should have received magick link via mail  by which you can validate the report without signup.

**Comment by ignacionelson** [comment]
Thank you. I've had a few issues accesing the links last week so I couldn't validate the issues. Will do it in order during the next few days.

**Comment by ranjit-git** [comment]
are you able to access the report?

**Comment by ranjit-git** [comment]
did you received the magick link to view report?

**Comment by kkplein** [comment]
Yes, an update here on the situation would appreciated.

**Comment by ignacionelson** [comment]
I've received the links and I'm working on 2 of the issues at the time

**Comment by ranjit-git** [comment]
@ignacionelson 
i have identified the vulnerable endpoint for your easy
https://huntr.dev/bounties/1625196531585-projectsend/projectsend/ 
---------------------------------------------------------------------
submitted a PR for fix the bug https://github.com/projectsend/projectsend/pull/971

https://huntr.dev/bounties/1625188746038-projectsend/projectsend/ -->
---------------------------------------------------------------------------------------
 vulnerable code at http://localhost/projectsend2/assets/js/app.js
                        ```
                        var li = $('<li/>')
                            .appendTo(list)
                            .html(`
                                <div class="date">
                                    <span>`+
                                        item.timestamp+`
                                    </span>
                                    <i class="fa fa-`+icon+`" aria-hidden="true"></i>
                                </div>
                                <div class="action">`+
                                    item.formatted+`             
                                </div>
                            `);
                            ```
                       Xss happening for varriable  ```item.formatted``` 
 to fix call htmlEncode(item.formatted) and htmlEncode function bellow 
        ```
        function htmlEncode(str){
               return String(str).replace(/[^\w. ]/gi, function(c){
               return '&#'+c.charCodeAt(0)+';';
               });
               }
               ```

https://huntr.dev/bounties/1625188855252-projectsend/projectsend/ 
-------------------------------------------------------------------
vulnerable code at http://localhost/projectsend2/assets/js/assets.js

```
<div class=\"plupload_file_name\"><span>'+o.name+\"</span></div>\"+'<div class=\"plupload_file_action\"><a href=\"#\"></a></div>'+'<div class=\"plupload_file_status\">
```
xss happening via ```o.name``` varriable
to fix the bug call htmlEncode(o.name) and htmlEncode function bellow
        ```
        function htmlEncode(str){
               return String(str).replace(/[^\w. ]/gi, function(c){
               return '&#'+c.charCodeAt(0)+';';
               });
               }
               ```

**Comment by ranjit-git** [comment]
@ignacionelson  can you plz  valided above all report ?
 i see fix deployment taking long time

**Comment by ignacionelson** [comment]
Just confirmed. Sorry for the delay, it's just me doing everything related to the project!
I have fixed the second one with your code here https://github.com/projectsend/projectsend/commit/1d90d547150daf0c115b371697e2ee0790218def, but since the commit is mine I don't know how to reward the bounty to you. Can you do a PR so I can credit you with the fix on huntr?

The third one comes from a dependency, so I'll have to check if plupload has this fixed on a newer version

**Comment by ranjit-git** [comment]
yes i see only you maintaining this repo. Dont worry i will help you to fix those.
For the second-one just created a patch for huntr .
And for 3rd bug i will report to pupload GH issue

**Comment by protego-it** [comment]
Any news here? What do we users have to do, to fix these problems? 
Thx in advance.

**Comment by malcolmp** [comment]
Hi @ignacionelson, I sent you an email regarding some potential help to resolve these issues.

**Comment by ignacionelson** [comment]
@protego-it please wait for a new release. I've been actively working on these issues

**Comment by ignacionelson** [comment]
@malcolmp will check it out now, thanks!

**Comment by kkplein** [comment]
Can we get an update here?

**Comment by ignacionelson** [comment]
Hi @kkplein 
Still working on this. Sadly I can only dedicate a few hours a week to PS so development time is not as fast as I'd like to.

**Comment by kkplein** [comment]
We understand your situation, and appreciate your hard work on PS!

**Comment by ranjit-git** [comment]
@ignacionelson 
Hello, i see it has been 6 month since bug reported  and still many of them are not validated .
As fix taking long time so you can validate the report now and when patch is ready  then you can confirm the fix also .
Huntr team did not proccessed the bounty to reporter untill it validated.
We invest our time to secure opensource project and report potential security vulnerability to huntr responsively .
If maintainer validate them then reporter gets bounty and it will encourage us to make opensource project a safer place .
Thanks

**Comment by ignacionelson** [comment]
Hi @ranjit-git 
I might have missed something, I thought all reports on huntr have been marked as fixed.
Could you please let me know the urls of those that are still open?
Thanks

**Comment by ranjit-git** [comment]
@ignacionelson  https://huntr.dev/bounties/1625240017884-projectsend/projectsend/
https://huntr.dev/bounties/1625229895180-projectsend/projectsend/
https://huntr.dev/bounties/1625229560780-projectsend/projectsend/
https://huntr.dev/bounties/1625229237062-projectsend/projectsend/
https://huntr.dev/bounties/1625228789050-projectsend/projectsend/
https://huntr.dev/bounties/1625228365696-projectsend/projectsend/
https://huntr.dev/bounties/1625228023793-projectsend/projectsend/
https://huntr.dev/bounties/1625227550260-projectsend/projectsend/
https://huntr.dev/bounties/1625198773635-projectsend/projectsend/
https://huntr.dev/bounties/1625197455990-projectsend/projectsend/
https://huntr.dev/bounties/1625195879034-projectsend/projectsend/
https://huntr.dev/bounties/1625188855252-projectsend/projectsend/

Thanks.

**Comment by ignacionelson** [comment]
Thanks. Will accept them and work on them ASAP.

**Comment by ranjit-git** [comment]
any update ? Few security report are still opened @ignacionelson

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
